pennylane/resource/specs.py

```diff
@@ -81,8 +123,75 @@ def circuit(x, add_ry=True):
     'gradient_fn': 'pennylane.gradients.parameter_shift.param_shift',
     'num_gradient_executions': 2}
 
+    Here you can see how the number of gates and their types change as we apply different amounts of transforms
+    through the ``level`` argument:
+
+    .. code-block:: python3
+
+        @qml.transforms.merge_rotations
+        @qml.transforms.undo_swaps
+        @qml.transforms.cancel_inverses
+        @qml.qnode(qml.device("default.qubit"), diff_method="parameter-shift", shifts=np.pi / 4)
+        def circuit(x):
+            qml.RandomLayers(qml.numpy.array([[1.0, 2.0]]), wires=(0, 1))
+            qml.RX(x, wires=0)
+            qml.RX(-x, wires=0)
+            qml.SWAP((0, 1))
+            qml.X(0)
+            qml.X(0)
+            return qml.expval(qml.X(0) + qml.Y(1))
+
+        return circuit
+
+    First, we can check the resource information of the ``QNode`` without any modifications. Note that ``level=top`` would
+    return the same results:
+
+    >>> qml.specs(circuit, level=0)(0.1)["resources"]
```

Comment:
```
    >>> qml.specs(circuit, level=0)(0.1)["resources"].gate_types
```

I wonder if it would enhance readability and communication if we just limited it to gate types.  Non-blocking comment though.